python/cudf/cudf/core/column/column.py

```diff
@@ -113,19 +113,63 @@ def as_frame(self) -> "cudf.core.frame.Frame":
             {None: self.copy(deep=False)}
         )
 
-    @property
-    def data_array_view(self) -> "cuda.devicearray.DeviceNDArray":
+    def data_array_view(
+        self, mode="write"
+    ) -> "cuda.devicearray.DeviceNDArray":
         """
         View the data as a device array object
+
+        Parameters
+        ----------
+        mode : str, default 'write'
+            Supported values are {'read', 'write'}
+            If 'write' is passed, a device array object
+            with readonly flag set to False in CAI is returned.
```

Comment:
Suggestion: make explicit what this means.

> If the caller wishes to modify the data returned through this view, they must pass mode="write".

Reply:
Done 👍